block_my_feedback.php

```diff
@@ -142,7 +142,7 @@ public function fetch_marking(stdClass $user): ?array {
                     $assess->cmid = $cmid;
                     $assess->modname = $mod->modname;
                     // Get due date and require marking.
-                    $assess = self::get_mod_data($mod, $assess);
+                    $assess = $this->get_mod_data($mod, $assess);
```

Comment:
Objects get passed by reference so we don't need to reassign `$assess`, i.e. we just need:

```
                    $this->get_mod_data($mod, $assess);
```

In which case `get_mod_data()` should be renamed to something like `populate_mod_data()` or `add_mod_data()` (because it's not a "getter").

Alternatively create `$assess` in `get_mod_data()` and return it, e.g.:

```
    public function fetch_marking(stdClass $user): ?array {
                    ⋮
                    $assess = $this->get_mod_data($mod);
                    ⋮
    public function get_mod_data($mod): ?stdClass {
                    $assess = new stdClass;
                    $assess->cmid = $cmid;
                    $assess->modname = $mod->modname;
                    ⋮
                    return $assess;
```

Reply:
@leonstr - could you expand on because it's not a "getter"?

Reply:
"Getters" and "setters" are typically wrappers for properties, e.g. `$o->set_name('A value')` and `$o->get_name()`. But one might use the get_... prefix for any function which returns a value.

But for a function with an object parameter, the object gets passed by reference ([technically an identifier](https://www.php.net/manual/en/language.oop5.references.php)) so you don't need to return the object to get the change, i.e.:

```
function f1($o): void {
    $o->modname = "test1";
}

function f2($o): stdClass {
    $o->modname = "test2";
    return $o;
}

$myobject = new stdClass();

f1($myobject);
echo $myobject->modname . "\n";

$myobject = f2($myobject);
echo $myobject->modname . "\n";
```

both do the same thing. Currently we have:

```
    public function get_mod_data($mod, $assess): ?stdClass {
⋮
                return $assess;
```

called with:

```
                    $assess = $this->get_mod_data($mod, $assess);
```

But we don't need to return `$assess` to access the added properties (`$assess->requiremarking`, etc.) we can just have:

```
                    $this->get_mod_data($mod, $assess);
```

and it works exactly the same because `$assess` gets directly modified by `get_mod_data()`.

So if (as I'm suggesting) we changed `get_mod_data()` to not return anything then the get_... prefix no longer makes sense. In which case I'd suggest changing this to `add_mod_data()`.

As always these are suggestions, please implement, defer or ignore as felt appropriate.

Apologies if there's anything here you already know.

Reply:
always appreciated :)